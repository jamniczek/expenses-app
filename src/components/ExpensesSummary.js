import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expensesTotal';
import numeral from 'numeral';

export const ExpensesSummary = ({ expenses }) => (
  <div>
    <h3>
      Viewing {expenses.length} expenses totalling{' '}
      {numeral(selectExpensesTotal(expenses) / 100).format('0.00')} PLN
    </h3>
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
