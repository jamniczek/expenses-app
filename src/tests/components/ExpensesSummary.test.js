import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary for 3 items', () => {
  const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary for 0 items', () => {
  const wrapper = shallow(<ExpensesSummary expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});
