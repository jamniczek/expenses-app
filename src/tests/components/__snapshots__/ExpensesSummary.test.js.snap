// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary for 0 items 1`] = `
<div>
  <h3>
    Viewing 
    0
     expenses totalling
     
    0.00
     PLN
  </h3>
</div>
`;

exports[`should render ExpensesSummary for 3 items 1`] = `
<div>
  <h3>
    Viewing 
    3
     expenses totalling
     
    2301.95
     PLN
  </h3>
</div>
`;
